### PR TITLE
Add wrapper for scikit-learn Pipelines

### PR DESCRIPTION
**Pull request by bukosabino**
https://github.com/Featuretools/featuretools/issues/163

### CODE REVIEW

**Comment by kmax12** [comment]
@bukosabino thank you!

looks like linting is causing tests to fail. can you run `make lint` locally until no errors come up? You make need to run `make installdeps` to install development dependencies first.

Currently, tests are failing with the following lint errors:

```
featuretools/tests/wrappers/test_sklearn_wrapper.py:18:37: E127 continuation line over-indented for visual indent
featuretools/tests/wrappers/test_sklearn_wrapper.py:19:37: E127 continuation line over-indented for visual indent
featuretools/tests/wrappers/test_sklearn_wrapper.py:22:39: E231 missing whitespace after ','
featuretools/tests/wrappers/test_sklearn_wrapper.py:25:5: E303 too many blank lines (2)
featuretools/tests/wrappers/test_sklearn_wrapper.py:29:33: E127 continuation line over-indented for visual indent
featuretools/tests/wrappers/test_sklearn_wrapper.py:34:57: E126 continuation line over-indented for hanging indent
featuretools/tests/wrappers/test_sklearn_wrapper.py:40:5: E303 too many blank lines (2)
featuretools/tests/wrappers/test_sklearn_wrapper.py:44:33: E127 continuation line over-indented for visual indent
featuretools/tests/wrappers/test_sklearn_wrapper.py:49:57: E126 continuation line over-indented for hanging indent
featuretools/tests/wrappers/test_sklearn_wrapper.py:55:5: E265 block comment should start with '# '
featuretools/tests/wrappers/test_sklearn_wrapper.py:55:5: E303 too many blank lines (2)
featuretools/tests/wrappers/test_sklearn_wrapper.py:63:5: E303 too many blank lines (2)
featuretools/tests/wrappers/test_sklearn_wrapper.py:66:32: E251 unexpected spaces around keyword / parameter equals
featuretools/tests/wrappers/test_sklearn_wrapper.py:66:34: E251 unexpected spaces around keyword / parameter equals
featuretools/tests/wrappers/test_sklearn_wrapper.py:67:32: E251 unexpected spaces around keyword / parameter equals
featuretools/tests/wrappers/test_sklearn_wrapper.py:67:34: E251 unexpected spaces around keyword / parameter equals
featuretools/tests/wrappers/test_sklearn_wrapper.py:75:5: E303 too many blank lines (2)
featuretools/wrappers/sklearn.py:183:28: E222 multiple spaces after operator
featuretools/wrappers/sklearn.py:220:24: E222 multiple spaces after operator
Makefile:9: recipe for target 'lint' failed
```

File: featuretools/tests/wrappers/test_sklearn_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from sklearn.ensemble import ExtraTreesClassifier
+from sklearn.model_selection import cross_val_score
+from sklearn.model_selection import GridSearchCV
+import pickle
+
+from featuretools.demo.mock_customer import load_mock_customer
+from featuretools.wrappers.sklearn import DFSTransformer
+
+
+def test_sklearn_wrapper():
```
Comment (kmax12) [inline featuretools/tests/wrappers/test_sklearn_wrapper.py]: can you split this method into multiple? each method should be named to describe what it does and only test one piece of functionality. 

if need look at how we use fixtures in other test files.

File: featuretools/wrappers/sklearn.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+            Calculates a feature matrix and features given a dictionary of entities and a list of relationships.
+
+            Args:
+                entity_ids (list): List of instances to calculate features on.
+
+            See Also:
+                :func:`synthesis.dfs`
+        """
+        self.feature_defs =    dfs(entities=self.entities,
```
Comment (kmax12) [inline featuretools/wrappers/sklearn.py]: because of `features_only`, you don't need the following arguments related to calculation in the fit() method: `save_progress`, `training_window`, `approximate`, `chunk_size`,` n_jobs`, `dask_kwargs`

File: featuretools/wrappers/sklearn.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+            Calculates a matrix for a given set of instance ids and calculation times.
+
+            Args:
+                entity_ids (list): List of instances to calculate features on.
+
+            See Also:
+                :func:`computational_backends.calculate_feature_matrix`
+        """
+        X_transformed =    calculate_feature_matrix(features=self.feature_defs,
```
Comment (kmax12) [inline featuretools/wrappers/sklearn.py]: `calculate_feature_matrix` isn't guaranteed to return in the same order as the instance ids provided. you should add 

`return X_transformed.loc[entity_ids]` to ensure the order is correct to this method

File: featuretools/wrappers/sklearn.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                                   features_only=True,
+                                   training_window=self.training_window,
+                                   approximate=self.approximate,
+                                   chunk_size=self.chunk_size,
+                                   n_jobs=self.n_jobs,
+                                   dask_kwargs=self.dask_kwargs,
+                                   verbose=self.verbose)
+        return self
+
+    def transform(self, entity_ids, y=None):
```
Comment (kmax12) [inline featuretools/wrappers/sklearn.py]: name `entity_ids` --> `instance_ids`

File: featuretools/wrappers/sklearn.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                                                    relationships=self.relationships,
+                                                    training_window=self.training_window,
+                                                    approximate=self.approximate,
+                                                    save_progress=self.save_progress,
+                                                    verbose=self.verbose,
+                                                    chunk_size=self.chunk_size,
+                                                    n_jobs=self.n_jobs,
+                                                    dask_kwargs=self.dask_kwargs,
+                                                    profile=self.profile)
+        return X_transformed
```
Comment (kmax12) [inline featuretools/wrappers/sklearn.py]: you also need to pass through `y` in the case where it's not None.

also, make sure we have a test case for providing a `y` and not providing a `y`

Comment (bukosabino) [reply featuretools/wrappers/sklearn.py]: Hi @kmax12,

Finally, I have decided don't use 'y' argument in 'transform' method. It does not make sense. Let me know if you agree.

**Comment by kmax12** [comment]
@bukosabino I think we need to support providing X as a list of cutoff times and not just instance ids. since the handling of time is one of featuretools key features. the logic would be: if it is a dataframe, pass to cutoff times argument. If it is a list or series pass to instance_ids argument.

File: featuretools/wrappers/sklearn.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                # Get examle data
+                es = ft.demo.load_mock_customer(return_entityset=True)
+                df = es['customers'].df
+
+                # Build dataset
+                pipeline = Pipeline(steps=[
+                    ('ft', dfsTransfomer(entityset=es,
+                                         target_entity="customers",
+                                         max_features=2)),
+                    ('sc', StandardScaler()),
```
Comment (kmax12) [inline featuretools/wrappers/sklearn.py]: can we add an classifier to this example since most people will want to use this wrapper to train a model? you can just make random labels to train it one like you do in the tests

then show calling `pipeline.predict()` at the end

**Comment by kmax12** [comment]
Looks good. Merging!
